Formalist/FormNumberFormatter.swift

```diff
@@ -0,0 +1,95 @@
+//
+//  FormNumberFormatter.swift
+//  Formalist
+//
+//  Created by Viktor Radchenko on 7/21/17.
+//  Copyright © 2017 Seed Platform, Inc. All rights reserved.
+//
+
+public enum FormNumberType {
+    case EIN
+    case SSN
+    case creditCard
+    case custom(pattern: String, replaceCharacter: Character)
+}
+
+public struct FormNumberFormatter: Formattable {
+
+    let type: FormNumberType
+
+    public init(
+        type: FormNumberType
+    ) {
+        self.type = type
+    }
+
+    public func from(input: String, previousInput: String = "") -> String {
+        let newValue = input.digits
+        var result = ""
+        let pattern = type.pattern
+        let replaceCharacter = type.replaceCharacter
+        var newValueIndex = newValue.startIndex
+        var patternIndex = pattern.startIndex
+        let isRemoving = newValue.characters.count < previousInput.characters.count
+
+        while patternIndex < pattern.endIndex && newValueIndex < newValue.endIndex  {
+            let value = pattern[patternIndex]
+
+            if value == replaceCharacter {
+                result.append(newValue[newValueIndex])
+                newValueIndex = newValue.index(newValueIndex, offsetBy: 1)
+            } else {
+                result.append(value)
+            }
+            patternIndex = pattern.index(patternIndex, offsetBy: 1)
+
+            if newValueIndex == newValue.endIndex && patternIndex < pattern.endIndex && pattern[patternIndex] != replaceCharacter {
+                if isRemoving {
```

Comment:
Yeah, tracking make this complicated to track, but it was nice to have this for UX, more intuitive to see mask, before you type. 

Implemented as you suggested for now, to reduce complexity.